types/tablet_v2/wlr_tablet_v2_tool.c

```diff
@@ -322,6 +322,15 @@ static void queue_tool_frame(struct wlr_tablet_tool_client_v2 *tool) {
 	}
 }
 
+static void handle_tablet_tool_surface_destroy(struct wl_listener *listener,
+		void *data) {
+	struct wlr_tablet_v2_tablet_tool *tool =
+		wl_container_of(listener, tool, surface_destroy);
+	wl_list_remove(&tool->surface_destroy.link);
+	wl_list_init(&tool->surface_destroy.link);
+	tool->focused_surface = NULL;
```

Comment:
Note for reviewer: I'm not really sure if this is enough, but it does fix the linked issue. In particular, `wlr_send_tablet_v2_tablet_tool_proximity_out` also clears `tool->current_client`, but I'm not sure if we should be doing that here, too.

Reply:
Hmm, yeah I think we should clear `current_client` too.

Should we just call `wlr_send_tablet_v2_tablet_tool_proximity_out`?

Reply:
Yeah, if we're also clearing `current_client` calling `proximity_out` makes sense to me too, it's already guarded by `current_client != NULL`.